ReactAndroid/src/main/java/com/facebook/react/uimanager/NativeViewHierarchyOptimizer.java

```diff
@@ -437,7 +437,7 @@ private void transitionLayoutOnlyViewToNativeView(
 
   private static boolean isLayoutOnlyAndCollapsable(@Nullable ReactStylesDiffMap props) {
     if (props == null) {
-      return true;
+      return false;
```

Comment:
Ok, I understand the problems you're having. Ideally we'd have some way to refer to the child views other than by index (since it isn't stable), but that doesn't exist right now and I don't know a good way to do it. I'm fine doing collapsableChildren false in this case then. (Actually, another option would be a create a custom StickyHeader native view class that ScrollView knows about and can do instanceof checks in native... but that would require a change in API)